srcs/http/response/http_method.cpp

```diff
@@ -36,63 +37,48 @@ std::string ReadFile(const std::string &file_path) {
 
 } // namespace
 
-// todo: 各メソッドを実行する関数
-// returnがある場合、設定する
-// - メソッドの権限の確認
-//  -> 権限がない場合405 Not Allowed
-// - 各メソッドを実行
-
 namespace http {
 
 StatusCode Method::Handler(
-	const std::string            &path,
-	const std::string            &method,
-	const std::list<std::string> &allow_method,
-	const std::string            &request_body_message,
-	std::string                  &response_body_message
+	const std::string  &path,
+	const std::string  &method,
+	const AllowMethods &allow_methods,
+	const std::string  &request_body_message,
+	std::string        &response_body_message
 ) {
 	StatusCode status_code(OK);
-	bool       is_allow_method = IsAllowedMethod(method, allow_method);
-	if (is_allow_method && method == GET) {
+	if (!IsAllowedMethod(method, allow_methods)) {
+		throw HttpException("Error: Not Implemented", StatusCode(NOT_IMPLEMENTED));
+	} else if (method == GET) {
 		status_code = GetHandler(path, response_body_message);
```

Comment:
`IsAllowedMethod` かそうじゃないか、で大きく 2 つに分かれるので、こういう場合は並列で  else if じゃなくて if から仕切り直しが自然かもです
```suggestion
	if (!IsAllowedMethod(method, allow_methods)) {
		throw HttpException("Error: Not Implemented", StatusCode(NOT_IMPLEMENTED));
	}
	if (method == GET) {
		status_code = GetHandler(path, response_body_message);
```